examples/auth/cert_auth_example.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Example for using client certificate authentication with CADC TAP service
+Tests for pyvo.extensions.authsession
+"""
+import requests
+
+import pyvo
+import pyvo.extensions.auth.authsession
+
+certificate_path = input('Path to client certificate file:')
+auth = pyvo.extensions.auth.authsession.AuthSession()
+auth.credentials.set_client_certificate(certificate_path)
+service = pyvo.dal.TAPService('https://www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/tap', auth)
+job = service.search('SELECT * from TAP_SCHEMA.tables')
+print(job)
```

Comment:
> auth can have a combination of multiple authorization methods and the class will try to match them with the ones supported by the service and create the appropriate internal sessions.
Right, that's what it does right now with that object, it matches the authorization methods against the ones supported by the service and uses the right session.

BasicAA is already supported by using the set_password function.  It uses the requests.sessions auth setting the tuple to username password.

> For extensions we could follow the requests model:
Yes, the idea is to allow people to put in their own sessions, with whatever authbase they want.  The problem I discovered in the previous version of this PR is that trying to have multiple authbases on one session is hard.  And some things like certificate auth are outside of the authbase part.  So making the session available is the safest bet to allow people to do whatever they want, including using whatever requests authenticators might be out there.

I'm not sure if these comments are calling for something, because this is already how it is in the code up for review?

Reply:
It was just an example to show that the API allows for other cases to be added later on.